### PR TITLE
Add option for type-keyword imports

### PR DESCRIPTION
**Pull request by lol768**
Before the change (with `"importsNotUsedAsValues": "error"` in `tsconfig.json` as e.g. the Vue.js project recommend), imports generated by this tool result in type errors:

- TS1371: This import is never used as a value and must use import type because importsNotUsedAsValues is set to error
- TS1444: BasketItem is a type and must be imported using a type-only import when preserveValueImports and isolatedModules are both enabled.

Equally, for folks on TS 5, `consistent-type-imports` from typescript-eslint would also complain about the imports that this package generates.

With a new `UseTypeImports` configuration flag on `TypeScriptGenerationOptions`, we can opt-in to using `type`-only imports where it makes sense (i.e. we're not trying to import a symbol/value):

Screenshot from 2023-11-06 12-22-38

Since this package never generates side-effect-reliant code, and just type definitions, it's safe to reference its generated types in this manner. To avoid breaking backwards-compat, I've left this off by default.

### CODE REVIEW

File: TypeScript.ContractGenerator/TypeScriptUnit.cs [hunk 1]
```diff
@@ -39,12 +40,13 @@ public TypeScriptVariableReference AddSymbolImport(string symbolName, string pat
...
                         CurrentUnit = this,
                         PathToUnit = path,
+                        UseTypeKeyword = false,
                     });
             }
             return new TypeScriptVariableReference(symbolName);
         }
 
-        public TypeScriptVariableReference AddDefaultSymbolImport(string localName, string path)
+        public TypeScriptVariableReference AddDefaultSymbolImport(string localName, string path, bool useTypeKeyword=false)
```
Comment (fakefeik) [inline TypeScript.ContractGenerator/TypeScriptUnit.cs:49]: AddDefaultSymbolImport and TypeScriptImportDefaultFromPathStatement is never used and can be removed

File: TypeScript.ContractGenerator/TypeScriptUnit.cs [hunk 2]
```diff
@@ -39,12 +40,13 @@ public TypeScriptVariableReference AddSymbolImport(string symbolName, string pat
                         TypeName = symbolName,
                         CurrentUnit = this,
                         PathToUnit = path,
+                        UseTypeKeyword = false,
```
Comment (fakefeik) [inline TypeScript.ContractGenerator/TypeScriptUnit.cs:43]: it's false by default, can be ommited

File: TypeScript.ContractGenerator/TypeScriptUnit.cs [hunk 3]
```diff
@@ -15,7 +15,7 @@ public class TypeScriptUnit
 
         public List<TypeScriptStatement> Body { get; } = new List<TypeScriptStatement>();
 
-        public TypeScriptTypeReference AddTypeImport(ITypeInfo sourceType, TypeScriptTypeDeclaration typeDeclaration, TypeScriptUnit sourceUnit)
+        public TypeScriptTypeReference AddTypeImport(ITypeInfo sourceType, TypeScriptTypeDeclaration typeDeclaration, TypeScriptUnit sourceUnit, bool useTypeKeyword=false)
```
Comment (fakefeik) [inline TypeScript.ContractGenerator/TypeScriptUnit.cs:18]: let's remove default parameter for useTypeKeyword, always pass parameter explicitly

File: TypeScript.ContractGenerator/Internals/TypeScriptImportFromUnitStatement.cs [hunk 4]
```diff
@@ -6,11 +6,13 @@ internal class TypeScriptImportFromUnitStatement : TypeScriptImportStatement
...
+            var typeKeyword = UseTypeKeyword ? "type " : "";
+            return $"import {typeKeyword}{{ {TypeName} }} from '{context.GetReferenceFromUnitToAnother(CurrentUnit.Path, TargetUnit.Path)}';";
         }
 
         public string TypeName { get; set; }
         public TypeScriptUnit TargetUnit { get; set; }
         public TypeScriptUnit CurrentUnit { get; set; }
+        public bool UseTypeKeyword { get; set; }
     }
-}
```
Comment (fakefeik) [inline TypeScript.ContractGenerator/Internals/TypeScriptImportFromUnitStatement.cs:16]: we also want codestyle tests to be green [here](https://github.com/skbkontur/TypeScript.ContractGenerator/actions/runs/6770752601/job/18485244783)

**Review by fakefeik — CHANGES_REQUESTED** [review]
thank you for cool PR, but it needs a bit of work

**Comment by lol768** [comment]
Thanks for the review @fakefeik - I think I've addressed everything, but give me a shout if there's anything outstanding.

**Review by fakefeik — APPROVED** [review]
looks good, will merge tomorrow

**Pull request merged**